rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6552,312 @@ private void assertRefundInProcessPegInVersionLegacy(
+        }
+
+        if (includeRetiringFederation) {
+            Address retiringFederationAddress = getFastBridgeAddressFromRedeemScript(
+                retiringFederation.getRedeemScript(),
+                Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+            );
+            tx.addOutput(valueToSend, retiringFederationAddress);
+        }
+        tx.addInput(
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes()),
+            0, ScriptBuilder.createInputScript(null, srcKey));
+
+        List<Sha256Hash> hashes = new ArrayList<>();
+        hashes.add(tx.getHash());
+        PartialMerkleTree pmt = new PartialMerkleTree(bridgeConstants.getBtcParams(), bits, hashes, 1);
+        Sha256Hash merkleRoot = pmt.getTxnHashAndMerkleRoot(new ArrayList<>());
+        co.rsk.bitcoinj.core.BtcBlock registerHeader = new co.rsk.bitcoinj.core.BtcBlock(
+            bridgeConstants.getBtcParams(),
+            1,
+            PegTestUtils.createHash(2),
+            merkleRoot,
+            1,
+            1,
+            1,
+            new ArrayList<>()
+        );
+        // simulate blockchain
+        mockChainOfStoredBlocks(
+            btcBlockStore,
+            registerHeader,
+            height + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations(),
+            height
+        );
+
+        InternalTransaction rskTx = new InternalTransaction(
+            Keccak256.ZERO_HASH.getBytes(),
+            0,
+            0,
+            null,
+            null,
+            null,
+            lbcAddress.getBytes(),
+            null,
+            null,
+            null,
+            null
+        );
+
+        co.rsk.core.Coin preCallLbcAddressBalance = repository.getBalance(lbcAddress);
+
+        BigInteger result = bridgeSupport.registerFastBridgeBtcTransaction(
+            rskTx,
+            tx.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize(),
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lbcAddress,
+            lpBtcAddress,
+            true
+        );
+
+        if (includeActiveFederation && !includeRetiringFederation) {
+            co.rsk.core.Coin postCallLbcAddressBalance = repository.getBalance(lbcAddress);
+            Assert.assertEquals(
+                preCallLbcAddressBalance.add(co.rsk.core.Coin.fromBitcoin(valueToSend)),
+                postCallLbcAddressBalance
+            );
+
+            verify(provider, times(1)).markFastBridgeFederationDerivationHashAsUsed(
+                tx.getHash(false),
+                fastBridgeDerivationHash
+            );
+
+            verify(provider, times(1)).setFastBridgeFederationInformation(
+                any()
+            );
+        }
+        return result;
+    }
+
+    @Test
+    public void registerFastBridgeBtcTransaction_before_RSKIP293_activation() throws IOException, BlockStoreException, BridgeIllegalArgumentException {
```

Comment:
All these cases could be written as different tests (i.e. in different methods) just to keep it organized. So each test name can tell us what it's testing. The way you did with the post activation tests